torchtune/utils/_profiler.py

```diff
@@ -48,3 +61,405 @@ def trace_handler(prof) -> None:
         if enabled
         else contextlib.nullcontext()
     )
+
+
+PROFILER_KEY = "profiler"
+_DEFAULT_PROFILER_ACTIVITIES = {
+    torch.profiler.ProfilerActivity.CPU,
+    torch.profiler.ProfilerActivity.CUDA,
+}
+
+_DEFAULT_SCHEDULE: dict = {
+    "wait": 5,
+    "warmup": 5,
+    "active": 2,
+    "repeat": 1,
+}
+
+DEFAULT_TRACE_OPTS: dict = {
+    "profile_memory": False,
+    "with_stack": False,
+    "record_shapes": True,
+    "with_flops": False,
+}
+
+_DEFAULT_PROFILE_DIR: str = "profiler_output"
+
+
+def _warn(msg: str):
+    _, rank = get_world_size_and_rank()
+    if rank == 0:
+        log.warn(msg)
+
+
+def trace_handler(
+    prof: torch.profiler.profiler.profile,
+    output_dir,
+    metric="self_cuda_time_total",
+    row_limit=25,
+):
```

Comment:
Do you mind adding a docstring for this? :)

Reply:
Done